packages/collector/test/tracing/database/ioredis/test.js

```diff
@@ -1281,4 +1281,142 @@ function checkConnection(span, setupType) {
+    describe('when ignore-endpoints is enabled via agent configuration', () => {
+      const { AgentStubControls } = require('../../../apps/agentStubControls');
+      const customAgentControls = new AgentStubControls();
+      let controls;
+
+      before(async () => {
+        await customAgentControls.startAgent({
+          ignoreEndpoints: { redis: 'get|set' }
+        });
+
+        controls = new ProcessControls({
+          agentControls: customAgentControls,
+          dirname: __dirname
+        });
+        await controls.startAndWaitForAgentConnection();
+      });
+
+      beforeEach(async () => {
+        await customAgentControls.clearReceivedTraceData();
+      });
+
+      after(async () => {
+        await customAgentControls.stopAgent();
+        await controls.stop();
+      });
+
+      it('should ignore redis spans for ignored endpoints (get, set)', async () => {
+        await controls
+          .sendRequest({
+            method: 'POST',
+            path: '/values',
+            qs: {
+              key: 'discount',
+              value: 50
+            }
+          })
+          .then(async () => {
+            return retry(async () => {
+              const spans = await customAgentControls.getSpans();
+              // 1 x http entry span
+              expect(spans.length).to.equal(1);
+              spans.forEach(span => {
+                expect(span.n).not.to.equal('redis');
+              });
+              expectAtLeastOneMatching(spans, [
+                span => expect(span.n).to.equal('node.http.server'),
+                span => expect(span.data.http.method).to.equal('POST')
+              ]);
+            });
+          });
+      });
+    });
+    describe('when ignore-endpoints is enabled via tracing configuration', async () => {
+      globalAgent.setUpCleanUpHooks();
+      const agentControls = globalAgent.instance;
+      let controls;
+
+      before(async () => {
+        controls = new ProcessControls({
+          useGlobalAgent: true,
+          dirname: __dirname,
+          env: {
+            INSTANA_IGNORE_ENDPOINTS: '{"redis": ["get"]}'
+          }
+        });
+        await controls.start();
+      });
+
+      beforeEach(async () => {
+        await agentControls.clearReceivedTraceData();
+      });
+
+      after(async () => {
+        await controls.stop();
+      });
+
+      afterEach(async () => {
+        await controls.clearIpcMessages();
+      });
+      it('should ignore spans for ignored endpoint (get)', async function () {
+        await controls
+          .sendRequest({
+            method: 'GET',
+            path: '/values',
+            qs: {
+              key: 'discount',
+              value: 50
+            }
+          })
+
+          .then(async () => {
+            return retry(async () => {
+              const spans = await agentControls.getSpans();
+              // 1 x http entry span
+              expect(spans.length).to.equal(1);
```

Comment:
issue: I am still missing the timeout. If you are expecting 1 span instead of 2. You need to wait to ensure you are not asserting a false positive.

Reply:
I have verified this, and it is not a false positive. Could you please help me understand what the issue is?

Reply:
So usually we do this in the test:
- send the request to the app
- start retry
- get spans
- wait til we have a number X of spans
- then do asserts

But because we ignore the endpoint, we only wait for 1 span instead of 2.
But if a second span would come in, we would not catch the problem. This is a false positive.

But maybe I am wrong, because the entry span might not get transmitted before the exits dont finish 🤔 🤔 🤔